auth/crammd5/authenticatee.go

```diff
@@ -0,0 +1,236 @@
+package crammd5
+
+import (
+	"fmt"
+	"sync"
+
+	"code.google.com/p/gogoprotobuf/proto"
+	log "github.com/golang/glog"
+	"github.com/mesos/mesos-go/auth"
+	mesos "github.com/mesos/mesos-go/mesosproto"
+	"github.com/mesos/mesos-go/messenger"
+	"github.com/mesos/mesos-go/upid"
+	"golang.org/x/net/context"
+)
+
+var (
+	pidLock        sync.Mutex
+	pid            int
+	supportedMechs map[string]struct{}
+)
+
+type statusType int
+
+const (
+	READY statusType = iota
+	STARTING
+	STEPPING
+	COMPLETED
+	FAILED
+	ERROR
+	DISCARDED
+)
+
+type authenticateeProcess struct {
+	*messenger.MesosMessenger
+	credential mesos.Credential
+	client     upid.UPID
+	status     statusType
+	initOnce   sync.Once
+	done       chan struct{}
+	result     error
+	ctx        context.Context
+	cancel     context.CancelFunc
+	mech       *mechanism
+	stepFn     stepFunc
+}
+
+func init() {
+	supportedMechs = make(map[string]struct{})
```

Comment:
`map[string]struct{}` is a simple way to implement a "set" of strings in go. The newer code maps `string` to `mech.Factory` so it's a bit more intuitive.